src/Sarif.FunctionalTests/SarifValidatorTests.cs

```diff
@@ -22,38 +23,47 @@ public class SarifValidatorTests
         public const string JsonSchemaFile = "Sarif.schema.json";
 
         private readonly string _jsonSchemaFilePath;
+        private readonly JsonSchema _schema;
 
         public SarifValidatorTests()
         {
             _jsonSchemaFilePath = Path.Combine(Environment.CurrentDirectory, JsonSchemaFile);
+            string schemaText = File.ReadAllText(JsonSchemaFile);
+            _schema = SchemaReader.ReadSchema(schemaText);
         }
 
         [Theory]
         [MemberData(nameof(DirectProducerTestCases))]
         public void Direct_Producer_validation(string inputFile)
         {
-            IEnumerable<JsonError> errors = Validator.ValidateFile(inputFile, JsonSchemaFile);
+            string instanceText = File.ReadAllText(inputFile);
+            var validator = new Validator(_schema);
+
+            string[] errors = validator.Validate(instanceText);
 
             // Test errors.Count(), rather than errors.Should().BeEmpty, because the latter
             // produces a less clear error message: it calls ToString on each member of
             // errors, and appends it to the string returned by FailureReason. Since
             // FailureReason already displayed the error messages in VisualStudio format,
             // there is no reason to append this additional, less well formatted information.
-            errors.Count().Should().Be(0, FailureReason(errors, inputFile));
+            errors.Count().Should().Be(0, FailureReason(errors));
         }
 
         [Theory]
         [MemberData(nameof(ConverterTestCases))]
         public void Converter_validation(string inputFile)
         {
-            IEnumerable<JsonError> errors = Validator.ValidateFile(inputFile, JsonSchemaFile);
+            string instanceText = File.ReadAllText(inputFile);
+            var validator = new Validator(_schema);
+
+            string[] errors = validator.Validate(instanceText);
```

Comment:
At present, the JSchema validator just returns an array of formatted error messages. When I modify the command line tool to emit SARIF, I'll probably need the validator to return an array of error objects -- possibly `Result` objects -- similar to what the VS-based validator did. (The VS-based validator returned an array of "`JsonError`" objects which the command-line tool turned into `Result` objects. But there's no reason the JSchema validator couldn't return `Result` objects directly.)